graphql_compiler/schema/sqlalchemy_schema.py

```diff
@@ -1,17 +1,92 @@
+
+    Args:
+        schema: GraphQLSchema
+        tables: dict mapping every graphql object type or interface type name in the schema to
+                a sqlalchemy table
+        junctions: dict mapping every graphql object type or interface type name in the schema to:
+                       dict mapping every vertex field name at that type to a dict with keys:
+                           from_column_name: string, column name to join from
+                           to_column_name: string, column name to join to
+        validate: Optional bool (default True), specifying whether to validate that the given
+                  input is valid for creation of a SQLAlchemySchemaInfo. Consider not validating
+                  to save on performance when dealing with a large schema.
+
+    Returns:
+        SQLAlchemySchemaInfo containing the input arguments provided
+    """
+    if validate:
+        types_mapped_to_tables = (GraphQLInterfaceType, GraphQLObjectType, GraphQLUnionType)
+        builtin_types = {
+            '__Schema',
+            '__Type',
+            '__Field',
+            '__InputValue',
+            '__EnumValue',
+            '__Directive',
+        }
+        builtin_fields = {
+            '_x_count',
+        }
+        for type_name, graphql_type in six.iteritems(schema.get_type_map()):
+            if isinstance(graphql_type, types_mapped_to_tables):
+                if type_name != 'RootSchemaQuery' and type_name not in builtin_types:
+                    # Check existence of sqlalchemy table for this type
+                    if type_name not in tables:
+                        raise AssertionError(u'Table for type {} not found'.format(type_name))
+                    table = tables[type_name]
+                    if not isinstance(table, sqlalchemy.Table):
+                        raise AssertionError(u'Table for type {} has wrong type {}'
+                                             .format(type_name, type(table)))
+
+                    # Check existence of all fields
+                    for field_name, field_type in six.iteritems(graphql_type.fields):
+                        is_vertex_field = (field_name.startswith('out_') or
+                                           field_name.startswith('in_'))
+                        if is_vertex_field:
+                            if field_name not in junctions[type_name]:
+                                raise AssertionError(u'No junction was specified for vertex '
+                                                     u'field {} on type {}'
+                                                     .format(field_name, type_name))
+                        else:
+                            # TODO(bojanserafimov): We have no SQL representation of list types yet,
+                            #                       so they are excused.
+                            # XXX should I just remove list types from the schema?
```

Comment:
Maybe raise `NotImplementedError()` on them for now? I wouldn't remove them altogether since I believe at least Postgres would allow list-typed columns, and we don't necessarily (yet) want to preclude supporting that in the future.

Reply:
I still want to be able to create an incomplete schema info, with array fields missing, so that I can run tests that don't rely on array fields.

Reply:
The way you'd implement aliases in MSSQL is to use a foreign key to a table of aliases. I'll implement it that way here, and raise `NotImplementedError()` when I need to deal with it later.

Reply:
See latest commit